src/mustachepkg/values.nim

```diff
@@ -24,13 +24,16 @@ type
     values: Table[string, Value]
     parent: Context
     searchDirs: seq[string]
+    partials: Table[string, string]
 
 proc derive*(val: Value, c: Context): Context;
 
-proc newContext*(searchDirs = @["./"]): Context =
-  Context(values: initTable[string,Value](), searchDirs: searchDirs)
+proc newContext*(searchDirs = @["./"], partials = initTable[string, string]()): Context =
+  Context(values: initTable[string,Value](), searchDirs: searchDirs, partials: partials)
 
 proc read*(c: Context, filename: string): string =
+  if c.partials.hasKey(filename):
```

Comment:
if I read this correctly, partials has precedence over searchDirs. Is this intended?

Reply:
Yes, as it is now, memory has precedence over searchDirs.  It was semi-intentional. I run in development using only `searchDirs` and in release build only using in-memory so I didn't ever have to decide which should really get precedence.  I'm fine with what @soasme describes below (explicitly specifying precedence).